model/convert/service_event.go

```diff
@@ -72,10 +72,14 @@ func convertServiceEventEpochSetup(event flow.Event) (*flow.ServiceEvent, error)
 	if !ok {
 		return nil, invalidCadenceTypeError("randomSource", cdcEvent.Fields[5], cadence.String(""))
 	}
-	setup.RandomSource, err = hex.DecodeString(string(randomSrcHex))
+	// Cadence's unsafeRandom().toString() produces a string of variable length.
```

Comment:
What is the cadence randomness here used for?

Reply:
The randomness is used as the seed for leader selection

Reply:
Hmm that random source is not safe and shouldn't be used for protocol stuff. Is there a simple to way the beacon output instead?
If not, this might be a good opportunity to upgrade Cadence `unsafeRandom` to use the random beacon.

Reply:
We are aware it is unsafe in general, but are OK to use this specifically for leader selection for the time being with the given network parameters. Once Cadence has access to the random beacon, we will update this, but it is not a blocker at the moment.